Extract License service from CCR and Watcher into license_api_guard plugin in x-pack

cjcenizal opened the pull request:
## Release note

Cross-cluster Replication and Watcher used to log misleading warnings about the basic license on Kibana startup. These warnings have been removed.

## Summary

Fixes https://github.com/elastic/kibana/issues/95032

When you're on the Basic license, Watcher and CCR log disconcerting license failure messages to the command line on startup. These messages are phrased in a way that makes users think something is wrong, when in reality nothing is wrong. This PR makes two fundamental changes:

* In terms of behavior, license failure messages are only logged when a consumer attempts to access a disallowed endpoint. Because the UI doesn't display apps that aren't covered by the user's license, such messages will no longer be logged on startup.
* In terms of architecture, the error messages we log are now localized within our internal license service instead of consumed from the core license service. In the long-term, they'll be removed from the core license service and plugins will be expected to follow this pattern and be responsible for their own license-failure UX.

I also refactored our internal license service to leverage the types exposed by the core license service.

Here's how license failures are now surfaced to the user in the UI and CLI for CCR:

![image](https://user-images.githubusercontent.com/1238659/113198546-e42ea800-921a-11eb-8a9c-9ea292837b97.png)

![image](https://user-images.githubusercontent.com/1238659/113198547-e5f86b80-921a-11eb-860c-a43620ae821a.png)

![image](https://user-images.githubusercontent.com/1238659/113198557-e7c22f00-921a-11eb-9396-11dc2fc04dfd.png)

![image](https://user-images.githubusercontent.com/1238659/113198564-ea248900-921a-11eb-8491-2cd87e718237.png)

![image](https://user-images.githubusercontent.com/1238659/113198567-ebee4c80-921a-11eb-869a-49a5687165dc.png)

Watcher is not quite as helpful but we can address that separately:

![image](https://user-images.githubusercontent.com/1238659/113462426-b477cf80-93d5-11eb-814d-9bff0a24bbf4.png)

![image](https://user-images.githubusercontent.com/1238659/113458834-82ac3c00-93c8-11eb-8d39-8b7cc21b69d8.png)

### Steps to test

* Edit the `license.ts` service to fake the license status.
* Load Watcher and CCR to send a request to the API and see how the response is surfaced in the UI and the Kibana server logs.

### Checklist

- [ ] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/master/packages/kbn-i18n/README.md)

elasticmachine commented:
Pinging @elastic/es-ui (Team:Elasticsearch UI)

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/116623/)
* Commit: 361d5014216d39b259afb2e8b3ee22917af2c1ed
* Storybooks not built yet
* This comment will update when the build is complete

### Failed CI Steps
* [Check i18n](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/116623/execution/node/453/log)

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/116623/","number":"116623","commit":"361d5014216d39b259afb2e8b3ee22917af2c1ed"}]}
PIPELINE-->

alisonelizabeth commented on x-pack/plugins/cross_cluster_replication/server/services/license.ts:
```diff
@@ -5,83 +5,103 @@
   ) {
+    this.logger = logger;
+
     licensing.license$.subscribe((license) => {
```
not directly related to this PR - but I noticed while looking at the code in my IDE that `LicensingPluginSetup.license$` is deprecated, and we should be using `license$` provided by the start contract.

cjcenizal replied:
Good point! Done.

alisonelizabeth reviewed (commented):
Thanks for working on this @cjcenizal! I’m happy with this approach and also verified the changes locally.

> Next steps: If reviewers approve of this direction, I'll extract this approach into a reusable service and then apply it to similar code in Watcher.

👍 I can re-review once this is complete. Can this service be used more broadly across our other plugins as well (perhaps addressed in a separate PR)?

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117130/)
* Commit: 4ba7a29a904e99b840896b55087db5fdc1eea27f
* Storybooks not built yet
* This comment will update when the build is complete

### Failed CI Steps
* [Setup Build Environment and Dependencies](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117130/execution/node/163/log)
* [Check Types](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117130/execution/node/449/log)
* [Setup Build Environment and Dependencies](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117130/execution/node/109/log)

### History
* :broken_heart: [Build #116623](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/116623/) failed 361d5014216d39b259afb2e8b3ee22917af2c1ed

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/116623/","number":"116623","commit":"361d5014216d39b259afb2e8b3ee22917af2c1ed"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117130/","number":"117130","commit":"4ba7a29a904e99b840896b55087db5fdc1eea27f"}]}
PIPELINE-->

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117132/)
* Commit: 5e573a1d1d6ee315bdcbe8f5f3dc9f7c088ff32d
* Storybooks not built yet
* This comment will update when the build is complete

### Failed CI Steps
* [Setup Build Environment and Dependencies](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117132/execution/node/122/log)
* [Check Types](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117132/execution/node/461/log)
* [Setup Build Environment and Dependencies](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117132/execution/node/163/log)

### History
* :broken_heart: [Build #117130](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117130/) failed 4ba7a29a904e99b840896b55087db5fdc1eea27f
* :broken_heart: [Build #116623](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/116623/) failed 361d5014216d39b259afb2e8b3ee22917af2c1ed

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/116623/","number":"116623","commit":"361d5014216d39b259afb2e8b3ee22917af2c1ed"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117130/","number":"117130","commit":"4ba7a29a904e99b840896b55087db5fdc1eea27f"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117132/","number":"117132","commit":"5e573a1d1d6ee315bdcbe8f5f3dc9f7c088ff32d"}]}
PIPELINE-->

cjcenizal commented on src/plugins/es_ui_shared/server/license/license.ts:
```diff
@@ -0,0 +1,118 @@
+  RequestHandlerContext,
+} from 'src/core/server';
+
+/* eslint-disable @kbn/eslint/no-restricted-paths */
```
@elastic/kibana-core Do we still need this linting rule?

pgayvallet replied:
For concrete imports, we definitely do, as it would plainly breaks on OSS distrib.

For types, it's not a 'technical' requirement, but until the basic plugins are moved out of xpack, we should stick with that rule.

mshustov replied:
AFAIK we don't release OSS distributable starting from v7.11 https://www.elastic.co/downloads/past-releases#kibana-oss

But I'd rather plugins didn't import from `x-pack`. Note that the code in `src/` and `x-pack` have different licenses, so it's easy to leak commercial code outside by importing `runtime` code accidentally.
Regarding the current use case: why `licensing` code lives in `src/` at all? `Licensing` compatible types can be declared in the current file, but maybe an `x-pack` plugin is a better place?

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117138/)
* Commit: 87c0a21766646d3838633b31e51f4519fe1984cd
* Storybooks not built yet
* This comment will update when the build is complete

### Failed CI Steps
* [Setup Build Environment and Dependencies](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117138/execution/node/163/log)
* [Check Types](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117138/execution/node/461/log)
* [Setup Build Environment and Dependencies](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117138/execution/node/122/log)

### History
* :broken_heart: [Build #117132](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117132/) failed 5e573a1d1d6ee315bdcbe8f5f3dc9f7c088ff32d
* :broken_heart: [Build #117130](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117130/) failed 4ba7a29a904e99b840896b55087db5fdc1eea27f
* :broken_heart: [Build #116623](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/116623/) failed 361d5014216d39b259afb2e8b3ee22917af2c1ed

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/116623/","number":"116623","commit":"361d5014216d39b259afb2e8b3ee22917af2c1ed"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117130/","number":"117130","commit":"4ba7a29a904e99b840896b55087db5fdc1eea27f"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117132/","number":"117132","commit":"5e573a1d1d6ee315bdcbe8f5f3dc9f7c088ff32d"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117138/","number":"117138","commit":"87c0a21766646d3838633b31e51f4519fe1984cd"}]}
PIPELINE-->

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117319/)
* Commit: 937ed8a0facb747265b21f16a27e77599a7affb1
* Storybooks not built yet
* This comment will update when the build is complete

### Failed CI Steps
* [Setup Build Environment and Dependencies](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117319/execution/node/161/log)
* [Check Types](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117319/execution/node/434/log)
* [Setup Build Environment and Dependencies](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117319/execution/node/121/log)

### History
* :broken_heart: [Build #117138](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117138/) failed 87c0a21766646d3838633b31e51f4519fe1984cd
* :broken_heart: [Build #117132](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117132/) failed 5e573a1d1d6ee315bdcbe8f5f3dc9f7c088ff32d
* :broken_heart: [Build #117130](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117130/) failed 4ba7a29a904e99b840896b55087db5fdc1eea27f
* :broken_heart: [Build #116623](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/116623/) failed 361d5014216d39b259afb2e8b3ee22917af2c1ed

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/116623/","number":"116623","commit":"361d5014216d39b259afb2e8b3ee22917af2c1ed"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117130/","number":"117130","commit":"4ba7a29a904e99b840896b55087db5fdc1eea27f"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117132/","number":"117132","commit":"5e573a1d1d6ee315bdcbe8f5f3dc9f7c088ff32d"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117138/","number":"117138","commit":"87c0a21766646d3838633b31e51f4519fe1984cd"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117319/","number":"117319","commit":"937ed8a0facb747265b21f16a27e77599a7affb1"}]}
PIPELINE-->

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117325/)
* Commit: 3c6acd7b99f49de3b6ef214267ea95f8506decdc
* Storybooks not built yet
* This comment will update when the build is complete

### Failed CI Steps
* [Setup Build Environment and Dependencies](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117325/execution/node/161/log)
* [Check Types](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117325/execution/node/441/log)
* [Setup Build Environment and Dependencies](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117325/execution/node/121/log)

### History
* :broken_heart: [Build #117319](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117319/) failed 937ed8a0facb747265b21f16a27e77599a7affb1
* :broken_heart: [Build #117138](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117138/) failed 87c0a21766646d3838633b31e51f4519fe1984cd
* :broken_heart: [Build #117132](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117132/) failed 5e573a1d1d6ee315bdcbe8f5f3dc9f7c088ff32d
* :broken_heart: [Build #117130](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117130/) failed 4ba7a29a904e99b840896b55087db5fdc1eea27f
* :broken_heart: [Build #116623](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/116623/) failed 361d5014216d39b259afb2e8b3ee22917af2c1ed

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/116623/","number":"116623","commit":"361d5014216d39b259afb2e8b3ee22917af2c1ed"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117130/","number":"117130","commit":"4ba7a29a904e99b840896b55087db5fdc1eea27f"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117132/","number":"117132","commit":"5e573a1d1d6ee315bdcbe8f5f3dc9f7c088ff32d"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117138/","number":"117138","commit":"87c0a21766646d3838633b31e51f4519fe1984cd"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117319/","number":"117319","commit":"937ed8a0facb747265b21f16a27e77599a7affb1"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117325/","number":"117325","commit":"3c6acd7b99f49de3b6ef214267ea95f8506decdc"}]}
PIPELINE-->

cjcenizal commented on src/plugins/es_ui_shared/server/license/license.ts:
```diff
@@ -0,0 +1,120 @@
+  start({ pluginId, minimumLicenseType, licensing }: StartSettings) {
+    licensing.license$.subscribe((license) => {
+      this.licenseType = license.type;
+      this.licenseCheckState = license.check(pluginId, minimumLicenseType!).state;
```
To testers: you can hardcode a value here, e.g. 'invalid', to test the behavior under various conditions.

cjcenizal commented:
@alisonelizabeth @jloleysens This is ready for your 👀 !

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117353/)
* Commit: 7c34572abe226c57f0b14c712d0dcfb7b12c0647
* Storybooks not built yet
* This comment will update when the build is complete

### Failed CI Steps
* [Check Types](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117353/execution/node/431/log)

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117353/","number":"117353","commit":"7c34572abe226c57f0b14c712d0dcfb7b12c0647"}]}
PIPELINE-->

kibanamachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117319/)
* Commit: 937ed8a0facb747265b21f16a27e77599a7affb1
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-95973/937ed8a0facb747265b21f16a27e77599a7affb1)
* [Pipeline Steps](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117319/flowGraphTable) (look for red circles / failed steps)
* [Interpreting CI Failures](https://www.elastic.co/guide/en/kibana/current/interpreting-ci-failures.html)

### Failed CI Steps
* [Setup Build Environment and Dependencies](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117319/execution/node/161/log)
* [Check Types](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117319/execution/node/434/log)
* [Setup Build Environment and Dependencies](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117319/execution/node/121/log)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
> ✅ unchanged

### History
* :broken_heart: [Build #117353](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117353/) failed 7c34572abe226c57f0b14c712d0dcfb7b12c0647

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117353/","number":"117353","commit":"7c34572abe226c57f0b14c712d0dcfb7b12c0647"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117319/","number":"117319","commit":"937ed8a0facb747265b21f16a27e77599a7affb1"}]}
PIPELINE-->

kibanamachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117325/)
* Commit: 3c6acd7b99f49de3b6ef214267ea95f8506decdc
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-95973/3c6acd7b99f49de3b6ef214267ea95f8506decdc)
* [Pipeline Steps](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117325/flowGraphTable) (look for red circles / failed steps)
* [Interpreting CI Failures](https://www.elastic.co/guide/en/kibana/current/interpreting-ci-failures.html)

### Failed CI Steps
* [Setup Build Environment and Dependencies](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117325/execution/node/161/log)
* [Check Types](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117325/execution/node/441/log)
* [Setup Build Environment and Dependencies](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117325/execution/node/121/log)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
> ✅ unchanged

### History
* :broken_heart: [Build #117319](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117319/) failed 937ed8a0facb747265b21f16a27e77599a7affb1
* :broken_heart: [Build #117353](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117353/) failed 7c34572abe226c57f0b14c712d0dcfb7b12c0647

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117353/","number":"117353","commit":"7c34572abe226c57f0b14c712d0dcfb7b12c0647"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117319/","number":"117319","commit":"937ed8a0facb747265b21f16a27e77599a7affb1"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117325/","number":"117325","commit":"3c6acd7b99f49de3b6ef214267ea95f8506decdc"}]}
PIPELINE-->

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117399/)
* Commit: 25bc327c48f094403b3d299a7c7cbd2ce73cc367
* Storybooks not built yet
* This comment will update when the build is complete

### Failed CI Steps
* [Check Types](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117399/execution/node/420/log)

### History
* :broken_heart: [Build #117353](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117353/) failed 7c34572abe226c57f0b14c712d0dcfb7b12c0647
* :broken_heart: [Build #117325](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117325/) failed 3c6acd7b99f49de3b6ef214267ea95f8506decdc
* :broken_heart: [Build #117319](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117319/) failed 937ed8a0facb747265b21f16a27e77599a7affb1

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117319/","number":"117319","commit":"937ed8a0facb747265b21f16a27e77599a7affb1"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117325/","number":"117325","commit":"3c6acd7b99f49de3b6ef214267ea95f8506decdc"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117353/","number":"117353","commit":"7c34572abe226c57f0b14c712d0dcfb7b12c0647"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117399/","number":"117399","commit":"25bc327c48f094403b3d299a7c7cbd2ce73cc367"}]}
PIPELINE-->

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117415/)
* Commit: f6002ccafabb6ac0e22aa54d38eb6b3792e19ef2
* Storybooks not built yet
* This comment will update when the build is complete

### Failed CI Steps
* [Check Types](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117415/execution/node/422/log)

### History
* :broken_heart: [Build #117399](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117399/) failed 25bc327c48f094403b3d299a7c7cbd2ce73cc367
* :broken_heart: [Build #117353](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117353/) failed 7c34572abe226c57f0b14c712d0dcfb7b12c0647
* :broken_heart: [Build #117325](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117325/) failed 3c6acd7b99f49de3b6ef214267ea95f8506decdc
* :broken_heart: [Build #117319](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117319/) failed 937ed8a0facb747265b21f16a27e77599a7affb1

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117319/","number":"117319","commit":"937ed8a0facb747265b21f16a27e77599a7affb1"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117325/","number":"117325","commit":"3c6acd7b99f49de3b6ef214267ea95f8506decdc"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117353/","number":"117353","commit":"7c34572abe226c57f0b14c712d0dcfb7b12c0647"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117399/","number":"117399","commit":"25bc327c48f094403b3d299a7c7cbd2ce73cc367"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117415/","number":"117415","commit":"f6002ccafabb6ac0e22aa54d38eb6b3792e19ef2"}]}
PIPELINE-->

pgayvallet reviewed (commented):
I know this may change soon, but I'm not at ease with the fact a xpack-dependent utility class importing types from `licensing` is exposed from an OSS plugin.

@mshustov wdyt?

jloleysens commented on src/plugins/es_ui_shared/server/license/license.ts:
```diff
@@ -0,0 +1,124 @@
+    });
+  }
+
+  getLicenseErrorMessage(licenseCheckState: LicenseCheckState): string {
```
nit; if this does not need to be public I would mark it as `private`.

cjcenizal replied:
Good idea!

jloleysens reviewed (commented):
Happy with this direction @cjcenizal ! It would be nice to see this code used across all of the relevant plugins per @alisonelizabeth 's suggestion, perhaps we can follow up with an issue for tracking progress of that implementation once this work is merged.

I did not test the functionality locally, but left a few comments :)

My opinion here: https://github.com/elastic/kibana/pull/95973/files?file-filters%5B%5D=.js&file-filters%5B%5D=.ts#r607718624

I agree that it makes most sense to have a service like this live in x-pack. ES UI should discuss moving x-pack specific functionality to an x-pack plugin but if we are only importing types for now we don't need to block this work on that.

Lastly, I think it would be good to bring the licensing service code in line with our other packages inside of `es_ui_shared` and move it to the `__packages_do_not_import__` folder and export functionality through a `licesnings.ts` file. See `errors.ts` for reference inside the errors folder.

cjcenizal commented on src/plugins/es_ui_shared/server/license/license.test.ts:
```diff
@@ -0,0 +1,105 @@
+        warn: jest.fn(),
+      };
+
+      const licensingMock = {
+        license$: {
+          subscribe: (callback: (config: unknown) => {}) =>
+            callback({
+              type: currentLicenseType,
+              check: (): { state: LicenseCheckState } => ({ state: licenseState }),
+              getFeature: () => ({}),
+            }),
+        },
+      };
```
Thanks for pointing me to that mock! I'll try out the sync observable approach.

cjcenizal commented:
@alisonelizabeth @jloleysens @pgayvallet @mshustov Thanks for your reviews! I've addressed your feedback. I agree that we should sequester `x-pack` related code together until we make a formal commitment to move this code into `src`, so I've moved the License module into a new x-pack plugin called `license_api_guard`. At some point in the future I think we could discuss either moving it into the `licensing` plugin if we think non-ESUI plugins would benefit from it, or into an x-pack-specific version of `es_ui_shared` if not.

@alisonelizabeth To your question about using this in our other plugins -- I just checked and I believe all of our other plugins are licensed under Basic, so there's no need to guard their API routes this way.

Reviewers, would you mind taking another look at sharing your thoughts? Thanks!

cjcenizal commented on x-pack/plugins/license_api_guard/server/license.test.ts:
```diff
@@ -0,0 +1,104 @@
+import { of } from 'rxjs';
+import type { KibanaRequest, RequestHandlerContext } from 'src/core/server';
+// eslint-disable-next-line @kbn/eslint/no-restricted-paths
+import { httpServerMock } from 'src/core/server/http/http_server.mocks';
```
@mshustov Do you think we could update `src/core/server` to export these helpers publicly?

mshustov replied:
They are available from `src/core/server/mocks`

cjcenizal replied:
Thanks, fixed in https://github.com/elastic/kibana/pull/97334

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/119493/)
* Commit: 066a69f27acdc99e25923333f228aa671ba6b020
* Storybooks not built yet
* This comment will update when the build is complete

### Failed CI Steps
* [Setup Build Environment and Dependencies](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/119493/execution/node/161/log)
* [Check Plugin List Docs](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/119493/execution/node/413/log)
* [Setup Build Environment and Dependencies](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/119493/execution/node/121/log)

### History
* :broken_heart: [Build #117415](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117415/) failed f6002ccafabb6ac0e22aa54d38eb6b3792e19ef2
* :broken_heart: [Build #117399](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117399/) failed 25bc327c48f094403b3d299a7c7cbd2ce73cc367
* :broken_heart: [Build #117353](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117353/) failed 7c34572abe226c57f0b14c712d0dcfb7b12c0647
* :broken_heart: [Build #117325](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117325/) failed 3c6acd7b99f49de3b6ef214267ea95f8506decdc
* :broken_heart: [Build #117319](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117319/) failed 937ed8a0facb747265b21f16a27e77599a7affb1

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117319/","number":"117319","commit":"937ed8a0facb747265b21f16a27e77599a7affb1"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117325/","number":"117325","commit":"3c6acd7b99f49de3b6ef214267ea95f8506decdc"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117353/","number":"117353","commit":"7c34572abe226c57f0b14c712d0dcfb7b12c0647"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117399/","number":"117399","commit":"25bc327c48f094403b3d299a7c7cbd2ce73cc367"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117415/","number":"117415","commit":"f6002ccafabb6ac0e22aa54d38eb6b3792e19ef2"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/119493/","number":"119493","commit":"066a69f27acdc99e25923333f228aa671ba6b020"}]}
PIPELINE-->

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/119768/)
* Commit: 48293ff96fbda349f31993a821967a0204a5b908
* Storybooks not built yet
* This comment will update when the build is complete

### Failed CI Steps
* [Check Plugin List Docs](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/119768/execution/node/406/log)

### History
* :broken_heart: [Build #119493](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/119493/) failed 066a69f27acdc99e25923333f228aa671ba6b020
* :broken_heart: [Build #117415](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117415/) failed f6002ccafabb6ac0e22aa54d38eb6b3792e19ef2
* :broken_heart: [Build #117399](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117399/) failed 25bc327c48f094403b3d299a7c7cbd2ce73cc367
* :broken_heart: [Build #117353](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117353/) failed 7c34572abe226c57f0b14c712d0dcfb7b12c0647
* :broken_heart: [Build #117325](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117325/) failed 3c6acd7b99f49de3b6ef214267ea95f8506decdc

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117325/","number":"117325","commit":"3c6acd7b99f49de3b6ef214267ea95f8506decdc"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117353/","number":"117353","commit":"7c34572abe226c57f0b14c712d0dcfb7b12c0647"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117399/","number":"117399","commit":"25bc327c48f094403b3d299a7c7cbd2ce73cc367"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117415/","number":"117415","commit":"f6002ccafabb6ac0e22aa54d38eb6b3792e19ef2"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/119493/","number":"119493","commit":"066a69f27acdc99e25923333f228aa671ba6b020"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/119768/","number":"119768","commit":"48293ff96fbda349f31993a821967a0204a5b908"}]}
PIPELINE-->

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/119800/)
* Commit: a0fe425d82a8cdc4510c2c2f19be32c508315511
* Storybooks not built yet
* This comment will update when the build is complete

### Failed CI Steps
* [Check Plugin List Docs](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/119800/execution/node/417/log)

### History
* :broken_heart: [Build #119768](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/119768/) failed 48293ff96fbda349f31993a821967a0204a5b908
* :broken_heart: [Build #119493](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/119493/) failed 066a69f27acdc99e25923333f228aa671ba6b020
* :broken_heart: [Build #117415](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117415/) failed f6002ccafabb6ac0e22aa54d38eb6b3792e19ef2
* :broken_heart: [Build #117399](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117399/) failed 25bc327c48f094403b3d299a7c7cbd2ce73cc367
* :broken_heart: [Build #117353](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117353/) failed 7c34572abe226c57f0b14c712d0dcfb7b12c0647

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117353/","number":"117353","commit":"7c34572abe226c57f0b14c712d0dcfb7b12c0647"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117399/","number":"117399","commit":"25bc327c48f094403b3d299a7c7cbd2ce73cc367"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117415/","number":"117415","commit":"f6002ccafabb6ac0e22aa54d38eb6b3792e19ef2"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/119493/","number":"119493","commit":"066a69f27acdc99e25923333f228aa671ba6b020"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/119768/","number":"119768","commit":"48293ff96fbda349f31993a821967a0204a5b908"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/119800/","number":"119800","commit":"a0fe425d82a8cdc4510c2c2f19be32c508315511"}]}
PIPELINE-->

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/119849/)
* Commit: 0706a2589e4c5bcf6c3ed06f4edb6da160bf8e91
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-95973/0706a2589e4c5bcf6c3ed06f4edb6da160bf8e91)
* [Documentation Changes](https://kibana_95973.docs-preview.app.elstc.co/diff)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
> ✅ unchanged

### History
* :broken_heart: [Build #119800](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/119800/) failed a0fe425d82a8cdc4510c2c2f19be32c508315511
* :broken_heart: [Build #119768](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/119768/) failed 48293ff96fbda349f31993a821967a0204a5b908
* :broken_heart: [Build #119493](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/119493/) failed 066a69f27acdc99e25923333f228aa671ba6b020
* :broken_heart: [Build #117415](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117415/) failed f6002ccafabb6ac0e22aa54d38eb6b3792e19ef2
* :broken_heart: [Build #117399](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117399/) failed 25bc327c48f094403b3d299a7c7cbd2ce73cc367

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117399/","number":"117399","commit":"25bc327c48f094403b3d299a7c7cbd2ce73cc367"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117415/","number":"117415","commit":"f6002ccafabb6ac0e22aa54d38eb6b3792e19ef2"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/119493/","number":"119493","commit":"066a69f27acdc99e25923333f228aa671ba6b020"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/119768/","number":"119768","commit":"48293ff96fbda349f31993a821967a0204a5b908"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/119800/","number":"119800","commit":"a0fe425d82a8cdc4510c2c2f19be32c508315511"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/119849/","number":"119849","commit":"0706a2589e4c5bcf6c3ed06f4edb6da160bf8e91"}]}
PIPELINE-->

mshustov commented on x-pack/plugins/license_api_guard/server/license.ts:
```diff
@@ -0,0 +1,115 @@
+  }
+
+  start({ pluginId, minimumLicenseType, licensing }: StartSettings) {
+    licensing.license$.subscribe((license: ILicense) => {
```
shouldn't we unsubscribe on the `stop` lifecycle?

cjcenizal replied:
@mshustov I was under the impression that `stop` is never called in server-side plugins -- plugins just "live" until the Kibana server process is killed. Is that not the case?

mshustov replied:
> plugins just "live" until the Kibana server process is killed.

Nope. It's always called during the normal teardown process (unless Kibana crashes).

cjcenizal replied:
When does teardown occur? How can I trigger it?

mshustov commented on x-pack/plugins/license_api_guard/server/license.ts:
```diff
@@ -0,0 +1,115 @@
+        const licenseErrorMessage = this.getLicenseErrorMessage(this.licenseCheckState);
+        this.logger?.warn(licenseErrorMessage);
+
+        return response.customError({
```
why not `response.forbidden`?
```js
        return response.forbidden({
          body: {
            message: licenseErrorMessage,
          },
        });
```

cjcenizal replied:
Good suggestion!

mshustov reviewed (commented):
> At some point in the future I think we could discuss either moving it into the licensing plugin if we think non-ESUI plugins would benefit from it, or into an x-pack-specific version of es_ui_shared if not.

We are planning to add something similar https://github.com/elastic/kibana/issues/56926

alisonelizabeth commented on x-pack/plugins/license_api_guard/server/license.ts:
```diff
@@ -0,0 +1,115 @@
+  }
+
+  // eslint-disable-next-line @typescript-eslint/explicit-member-accessibility
+  get isEsSecurityEnabled() {
```
Can we remove the `// eslint-disable`?

```suggestion
  public get isEsSecurityEnabled() {
```

alisonelizabeth reviewed (approved):
Changes LGTM. Thanks @cjcenizal!

> @alisonelizabeth To your question about using this in our other plugins -- I just checked and I believe all of our other plugins are licensed under Basic, so there's no need to guard their API routes this way.

I think most - if not all - of our plugins have a similar license check (for example, [Ingest Node Pipelines](https://github.com/elastic/kibana/blob/master/x-pack/plugins/ingest_pipelines/server/services/license.ts), [Index Management](https://github.com/elastic/kibana/blob/master/x-pack/plugins/index_management/server/services/license.ts) and [ILM](https://github.com/elastic/kibana/blob/master/x-pack/plugins/index_lifecycle_management/server/services/license.ts)). Can we remove this guard now?

jloleysens reviewed (approved):
Thanks for addressing that feedback @cjcenizal. Happy with the current changes!

cjcenizal commented:
@alisonelizabeth Great call, I created https://github.com/elastic/kibana/issues/97314 to track this work. @mshustov I'm going to go ahead and merge once CI is green. We can follow up on your comment about unsubscribing from the license service in another PR if this needs to be addressed.

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/120148/)
* Commit: d54f85cadac6eaec4bac92238cf7736ec0867696
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-95973/d54f85cadac6eaec4bac92238cf7736ec0867696)
* [Documentation Changes](https://kibana_95973.docs-preview.app.elstc.co/diff)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
> ✅ unchanged

### History
* :green_heart: [Build #119849](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/119849/) succeeded 0706a2589e4c5bcf6c3ed06f4edb6da160bf8e91
* :broken_heart: [Build #119800](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/119800/) failed a0fe425d82a8cdc4510c2c2f19be32c508315511
* :broken_heart: [Build #119768](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/119768/) failed 48293ff96fbda349f31993a821967a0204a5b908
* :broken_heart: [Build #119493](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/119493/) failed 066a69f27acdc99e25923333f228aa671ba6b020
* :broken_heart: [Build #117415](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117415/) failed f6002ccafabb6ac0e22aa54d38eb6b3792e19ef2

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117415/","number":"117415","commit":"f6002ccafabb6ac0e22aa54d38eb6b3792e19ef2"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/119493/","number":"119493","commit":"066a69f27acdc99e25923333f228aa671ba6b020"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/119768/","number":"119768","commit":"48293ff96fbda349f31993a821967a0204a5b908"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/119800/","number":"119800","commit":"a0fe425d82a8cdc4510c2c2f19be32c508315511"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/119849/","number":"119849","commit":"0706a2589e4c5bcf6c3ed06f4edb6da160bf8e91"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/120148/","number":"120148","commit":"d54f85cadac6eaec4bac92238cf7736ec0867696"}]}
PIPELINE-->

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/120196/)
* Commit: d54f85cadac6eaec4bac92238cf7736ec0867696
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-95973/d54f85cadac6eaec4bac92238cf7736ec0867696)
* [Documentation Changes](https://kibana_95973.docs-preview.app.elstc.co/diff)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
> ✅ unchanged

### History
* :green_heart: [Build #120148](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/120148/) succeeded d54f85cadac6eaec4bac92238cf7736ec0867696
* :green_heart: [Build #119849](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/119849/) succeeded 0706a2589e4c5bcf6c3ed06f4edb6da160bf8e91
* :broken_heart: [Build #119800](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/119800/) failed a0fe425d82a8cdc4510c2c2f19be32c508315511
* :broken_heart: [Build #119768](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/119768/) failed 48293ff96fbda349f31993a821967a0204a5b908
* :broken_heart: [Build #119493](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/119493/) failed 066a69f27acdc99e25923333f228aa671ba6b020

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/119493/","number":"119493","commit":"066a69f27acdc99e25923333f228aa671ba6b020"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/119768/","number":"119768","commit":"48293ff96fbda349f31993a821967a0204a5b908"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/119800/","number":"119800","commit":"a0fe425d82a8cdc4510c2c2f19be32c508315511"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/119849/","number":"119849","commit":"0706a2589e4c5bcf6c3ed06f4edb6da160bf8e91"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/120148/","number":"120148","commit":"d54f85cadac6eaec4bac92238cf7736ec0867696"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/120196/","number":"120196","commit":"d54f85cadac6eaec4bac92238cf7736ec0867696"}]}
PIPELINE-->
